NDB.Covid19/NDB.Covid19.Droid/Views/AuthenticationFlow/LoadingPageActivity.cs

```diff
@@ -74,16 +69,21 @@ void OnActivityFinished()
             
         }
 
-        void OnError(AccessDeniedException e)
-        {
-             NavigationHelper.GoToResultPageAndClearTop(this);
-        }
-
         void OnError(Exception e)
         {
-            AuthErrorUtils.GoToTechnicalError(this,LogSeverity.ERROR, e, "Pushing keys failed" );
+            if (e is AccessDeniedException)
+            {
+                LogUtils.LogMessage(LogSeverity.INFO, "The user refused to share keys", null);
+                GoToInfectionStatusPage();
+            }
+            else
+            {
+                RunOnUiThread(() => AuthErrorUtils.GoToTechnicalError(this, LogSeverity.ERROR, e, "Pushing keys failed"));
```

Comment:
probably goToInfectionStatusPage should also be done on UI thread

Reply:
@nc-kano Should Jonas do anything in this regard, or will we just merge this PR into dev?

Reply:
yes, just needs to add RunOnUIThread :)

Reply:
@jonasofrich see above :)

Reply:
Done